pkg/storage/util.go

```diff
@@ -56,6 +58,10 @@ func NamespaceKeyFunc(prefix string, obj runtime.Object) (string, error) {
 	if err != nil {
 		return "", err
 	}
+	name := meta.Name()
+	if ok, msg := validation.ValidatePathSegmentName(name, false); !ok {
```

Comment:
All of these can be replaced with pkg/util/validation.IsDNS1123Subdomain(name) (or a local wrapper thereof) and should probably be commented as "sanity checks".